spec/features/welcome_spec.rb

```diff
@@ -8,20 +8,44 @@
 
   scenario 'the visitor can input text and search result number and search' do
     visit '/'
-    fill_in('search_term_input',  with: 'water')
-    fill_in('result_count', with: '50')
+    fill_in('search_term',  with: 'water')
+    fill_in('count', with: '50')
     click_button 'Search'
     expect(page).to have_text("water")
-    within("div.ab-search-wrapper-left") do
-      expect(page).to have_css("div.changeless-box", count: 50)
+    within("table.results-table") do
+      expect(page).to have_css("tr.table-row", count: 52)
     end
   end
 
   scenario 'when "search" is clicked without search term top 10 results are displayed' do
     visit '/'
     click_button 'Search'
-    within("div.ab-search-wrapper-left") do
-      expect(page).to have_css("div.changeless-box", count: 10)
+    within("table.results-table") do
+      expect(page).to have_css("tr.table-row", count: 12)
+      # => 10 rows for the results, one header and a result count at the bottom
     end
   end
+
+  scenario 'the search function accepts search terms of different lengths' do
+    visit '/'
+    fill_in('search_term',  with: 'apple gds')
+    fill_in('count', with: '50')
+    expect(page.status_code).to eq(200)
+
+    visit '/'
```

Comment:
This feels a little overkill, it's not a big issue while the codebase is small, however as it grows have large/slow test will make the project harder to work on. As such it is better to aim for have as quick a set of tests as possible.

Further to this if you abstract the comparison to a separate class that would allow you to unit test the system, reducing the need for this type of large integration test. 😄